jolt-core/src/subprotocols/sumcheck.rs

```diff
@@ -416,13 +395,11 @@ impl<F: PrimeField> SumcheckInstanceProof<F> {
                     let len = poly_A.len() / 2;
 
                     // In the case of a flagged tree, the majority of the leaves will be 1s, optimize for this case.
-                    let (eval_point_0, eval_point_2, eval_point_3) = (0..len).into_iter()
+                    let (eval_point_0, eval_point_2, eval_point_3) = (0..len).into_par_iter()
```

Comment:
I think we want to pick either parallelism across the batch or parallelism across the MLE, but not both. When I benched this I believe that parallelism across the batch performed better but it was when we had the randomness bug meaning the trees were *vast* majority 1s.

Reply:
in my testing, I found that using parallelism in both places had the best performance (in both isolated sumcheck benchmarks and e2e-except-r1cs)